### ISSUE TITLE
Custom dataset using jsonl containing <s>[INST]question[/INST]answer

### ISSUE THREAD

**Issue by TiagoJacobs** [meta-llama/llama-recipes#425]
Hello, I would like to perform a finetuning of Llama-2-7b with the finetunning.py recipe of this repository.

Here is the command:

```
python -m llama_recipes.finetuning --dataset "custom_dataset" --custom_dataset.file "/notebooks/mydataset.py"  --use_peft --one_gpu --peft_method lora --quantization --model_name $BASE_MODEL --output_dir $NEW_MODEL
```

As far as I understand, `mydataset.py` must have a function named `get_custom_dataset`, and this function should return a PyTorch style dataset, containing `input_ids`, `attention_mask` and `labels`.

Given that I have instructions in the below format (.jsonl file), where can I find an example of how to convert it to the expected format?

```
{"text": "<s>[INST] What features does BigBlueButton offer for online learning? [/INST] BigBlueButton offers features such as real-time sharing of audio, video, slides, chat, and screen, along with the ability for teachers to engage students with polling, breakout rooms, and whiteboard annotation. </s>"}
{"text": "<s>[INST] How does BigBlueButton support accessibility? [/INST] BigBlueButton supports accessibility through features like screen reader support, keyboard navigation, and live closed captioning, making it more accessible to users with disabilities. </s>"}
{"text": "<s>[INST] What are breakout rooms in BigBlueButton? [/INST] Breakout rooms in BigBlueButton allow instructors to split the participants into smaller groups for collaborative activities or discussions, with the ability to visit and monitor each room. </s>"}
{"text": "<s>[INST] Can BigBlueButton integrate with Learning Management Systems (LMS)? [/INST] Yes, BigBlueButton can integrate with various Learning Management Systems (LMS) like Moodle, Sakai, and Canvas, providing seamless access within those platforms. </s>"}
```

**Comment by HamidShojanazeri** [comment]
I wonder if this would be helpful https://github.com/meta-llama/llama-recipes/blob/main/recipes/finetuning/datasets/custom_dataset.py#L13 ?

**Comment by TiagoJacobs** [comment]
Hello @HamidShojanazeri - if there is no official demo, could I send a PR adding it?

I think that the custom dataset example should be simpler, closer to what a beginner (the target audience of a "recipes" project) would use.

Before investing time sending a PR, I would like to know if simplifying the fine tunning examples are compatible with the project roadmap.

Does it make sense?

**Comment by HamidShojanazeri** [comment]
@TiagoJacobs thanks for the feedback, can you please elaborate a bit on the goal, is it to have like simple instruct-finetuning example that can be more educational. 

For sure we would love your contribution to have more simpler examples that can be used in the educational space. Happy to work with you and land the PR. It would be awesome if that could be a new example/ something not covered already. Thanks,
